### DISCUSSION TITLE
Having issues Decrypting Binary File

### DISCUSSION THREAD

**Discussion by andrewlaguna824** [ProtonMail/gopenpgp#203] [Q&A]
Hello!

I have an encrypted, signed file that is in binary format. My goal is to decrypt the message and verify its signature.

Currently, I am using `helper.DecryptVerifyMessageArmored(senderPublicKey, receiverPrivateKey, nil, cipherText)`.

The cipher text is in binary format (e.g. `��`Կ�ú�K斖��1wPb"�jItc~��` ... ).

When calling the above decryption function, I receive the error:
```
 gopenpgp: unable to unarmor ciphertext: gopenpgp: error in unarmoring message: gopenpgp: unable to unarmor: EOF
```

After some digging around online, it seems that perhaps the issue is that the cipher text is unarmored ([link](https://github.com/openpgpjs/openpgpjs/issues/518)). 

If this is the issue, is there a way I need to armor the cipher text before passing to the decrypt function? I do not have the ability to change the incoming file -- it is being received from a third party.

As a note, I can successfully decrypt and verify the message using the `gpg` command line util.

**Comment by marinthiercelin** [comment]
Hi, there's no helper function to decrypt unarmored messages.

But you can write your own using the crypto package.

The code would look something like that:

```go
func DecryptMessageUnarmored(
	privateKey string, passphrase []byte, ciphertext []byte,
) (plaintext string, err error) {
	var privateKeyObj, unlockedKeyObj *crypto.Key
	var privateKeyRing *crypto.KeyRing
	var pgpMessage *crypto.PGPMessage
	var message *crypto.PlainMessage

	if privateKeyObj, err = crypto.NewKeyFromArmored(privateKey); err != nil {
		return "", errors.Wrap(err, "gopenpgp: unable to unarmor private key")
	}

	if unlockedKeyObj, err = privateKeyObj.Unlock(passphrase); err != nil {
		return "", errors.Wrap(err, "gopenpgp: unable to unlock private key")
	}
	defer unlockedKeyObj.ClearPrivateParams()

	if privateKeyRing, err = crypto.NewKeyRing(unlockedKeyObj); err != nil {
		return "", errors.Wrap(err, "gopenpgp: unable to create new keyring")
	}

	if pgpMessage, err = crypto.NewPGPMessage(ciphertext); err != nil {
		return "", errors.Wrap(err, "gopenpgp: unable to unarmor ciphertext")
	}

	if message, err = privateKeyRing.Decrypt(pgpMessage, nil, 0); err != nil {
		return "", errors.Wrap(err, "gopenpgp: unable to decrypt message")
	}

	return message.GetString(), nil
}
```

This is basically the same helper, but the message object is built with `crypto.NewPGPMessage` instead of `crypto.NewPGPMessageFromArmored`

**Reply by andrewlaguna824** [reply]
Thank you for the quick response! I received the below error now:
`gopenpgp: error in reading message: openpgp: unsupported feature: Symmetrically encrypted packets without MDC are not supported`

1. it looks like the sender is providing an encrypted packet with an old encryption algorithm
2. It seems odd that the error is stating this is a symmetrically encrypted packets, even though we're using asymmetric keys.

For 1. it looks like I may need to use the crypto library to get access to the Config that has `InsecureAllowUnauthenticatedMessages` as it doesn't look like this fork includes that. Would that be your suggestion?

**Reply by twiss** [reply]
OpenPGP messages always use symmetric encryption, (usually) in combination with asymmetric encryption.

This library is not a fork of go-crypto, it's a high-level wrapper, and as such doesn't provide support for the low-level config options, indeed. So I would suggest using go-crypto directly, in this case.

**Reply by andrewlaguna824** [reply]
Thank you for the response! I appreciate you and @marinthiercelin answering my questions.
